core-bundle/assets/controllers/toggle-fieldset-controller.js

```diff
@@ -0,0 +1,123 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    static values = {
+        id: String,
+        table: String,
+    }
+
+    static classes = ['collapsed'];
+
+    static afterLoad (identifier, application) {
+        const addController = (el, id, table) => {
+            const fs = el.parentNode;
+
+            fs.dataset.controller = `${fs.dataset.controller || ''} ${identifier}`;
+            fs.setAttribute(`data-${identifier}-id-value`, id);
+            fs.setAttribute(`data-${identifier}-table-value`, table);
+            fs.setAttribute(`data-${identifier}-collapsed-class`, 'collapsed');
+            el.setAttribute('data-action', `click->${identifier}#toggle`);
+        }
+
+        const migrateLegacy = () => {
+            document.querySelectorAll('legend[data-toggle-fieldset]').forEach(function(el) {
+                console && console.warn(`Using "data-toggle-fieldset" attribute on fieldset legend is deprecated and will be removed in Contao 6. Apply the "${identifier}" Stimulus controller instead.`);
+
+                const { id, table } = JSON.parse(el.getAttribute('data-toggle-fieldset'));
+                addController(el, id, table);
+            });
+
+            AjaxRequest.toggleFieldset = (el, id, table) => {
+                const fs = el.parentNode;
+
+                // already clicked, Stimulus controller was added dynamically
+                if (application.getControllerForElementAndIdentifier(fs, identifier)) {
+                    return;
+                }
+
+                window.console && console.warn('Using AjaxRequest.toggleFieldset is deprecated and will be removed in Contao 6. Apply the Stimulus actions instead.');
+
+                addController(el, id, table);
+
+                // optimistically wait until Stimulus has registered the new controller
+                setTimeout(() => {
+                    application.getControllerForElementAndIdentifier(fs, identifier).toggle();
+                }, 100);
+            };
+        }
+
+        // called as soon as registered so DOM may not have loaded yet
+        if (document.readyState === "loading") {
+            document.addEventListener("DOMContentLoaded", migrateLegacy)
+        } else {
+            migrateLegacy();
+        }
+    }
+
+    connect () {
+        if (this.element.querySelectorAll('label.error, label.mandatory').length) {
+            this.element.classList.remove(this.collapsedClass);
+        } else if (this.element.classList.contains('hide')) {
+            console && console.warn(`Using "hide" class on fieldset is deprecated and will will be removed in Contao 6. Use "${this.collapsedClass}" instead.`);
```

Comment:
Is it necessary to change this? The class comes from the DCA and if we change it, everyone has to adjust their legends, don‘t they?

<img width="917" alt="" src="https://github.com/leofeyer/contao/assets/1192057/9cdbcf7d-1049-48fa-81cf-78b440cf40e2">

Reply:
can you point to the code that applies this class? I was assuming this sets the state and not just adds a class...

Reply:
found and fixed in b42547598ae921c380d8a79d005a9fd61d3d4a9c

Reply:
I don‘t understand why we have to deprecate the class. It is a well-documented feature that everyone is using and I don‘t see how using `:collapsed` is any better. @contao/developers WDYT?

Reply:
I also don't think we should change this. What exactly is the issue with `hide` anyway?

Reply:
I also don't understand it but maybe there's a good reason. For me, `foobar:hide` is just the "default" state of this legend. After that, the back end session contains whatever you as a user want to have opened/closed.

Reply:
it is not deprecated? There's an explicit handling for `hide` in https://github.com/leofeyer/contao/commit/b42547598ae921c380d8a79d005a9fd61d3d4a9c#diff-fb74744bd45d8f0f5c3207987a6ae2d1456f2dc7f06467cb14c133bafe18b927R172, but I don't think we need to support any other dynamic class auto-handlig?

Reply:
Ah I see. This is when you set `class="hide"` manually for some reason. This is not related to the DCA palette.

Reply:
Well the core previously set `class="hide"` when you had `foobar_legend:hide` in your DCA. But I don't think that makes sense, `hide` is a keyword and not a class we should dynamically replace in the javascript to collapse a section.

Reply:
I still don't understand why using a class named `hide` is problematic? I frequently use classes like `show` or `hide`. Is there an issue I am not aware of?

Reply:
But why not re-use the `hide` class regardless?

Reply:
re-use? You mean rewrite everything from `collapsed` to `hide`? Sounds compeltely unrelated to this PR 😂 but would be a data-attribute in the new Stimulus controller, so an easy change 😎